src/backend/distributed/utils/resource_lock.c

```diff
@@ -122,6 +124,90 @@ lock_shard_resources(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * LockShardListResourcesOnFirstWorker acquires the resource locks for the specified
```

Comment:
I think it'd be useful to explain the reasoning behind acquiring locks on the first worker here.